algosdk/atomic_transaction_composer.py

```diff
@@ -239,6 +239,10 @@ def add_method_call(
             raise error.AtomicTransactionComposerError(
                 "AtomicTransactionComposer cannot exceed MAX_GROUP_SIZE transactions"
             )
+
+        if app_id is None:
```

Comment:
This argument types for this function don't allow `app_id` to be None. Since that's the case I'm not sure that #431 is an issue.

Do you `app_id` _should_ be allowed to be None?

Reply:
I don't but I can successfully pass None without it whining anywhere

Reply:
Perhaps we should make it throw an error if None is passed?

Reply:
That could also work, as long as the caller doesn't get a sneaky 0 in app args

Reply:
To confirm, you are seeing the right behavior for foreign app args when 0 is passed in for the app ID during create?

If so, based on that and the type annotations, I'm pretty sure the intended path is for users to explicitly pass 0 in for the app ID and we can consider passing None to be an error.